copyright_check

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+# Call copyright_check with 1 or more filepath arguments to check.
+
+copyright_header="\
+# (C) Crown Copyright, Met Office. All rights reserved.\n\
+#\n\
+# This file is part of 'dagrunner' and is released under the BSD 3-Clause license.\n\
+# See LICENSE in the root of the repository for full licensing details.\
+"
+
+contains_copyright() {
+    grep -qi '# .*copyright' $1
+}
+
+
+defines_hash_bang() {
+    grep -q '^#!' $1
+}
+
+
+correct_copyright() {
+    python -c 'import sys; sys.exit(0 if "'"$1"'" in open("'"$2"'", "r").read() else 1)'
+}
+
+found_error=0
+for filepath in "$@"; do
+    if contains_copyright "${filepath}"; then
+        #echo "File '${filepath}' already contains Crown Copyright"
+        if correct_copyright "${copyright_header}" "${filepath}"; then
+            continue
+        else
+            echo "Incorrect Copyright header in '${filepath}'"
+            found_error=1
+        fi
+    elif [ -s "${filepath}" ]; then  # skip empty files
+        echo "Adding missing Copyright to '${filepath}'"
+        tmp_file="${filepath}.tmp"
+        rm -f ${tmp_file} 2&> /dev/null
+        touch ${tmp_file}
+        if defines_hash_bang "${filepath}"; then
+            head -n 1 ${filepath} > ${tmp_file}
+            echo -e ${copyright_header} >> ${tmp_file}
+            tail -n +2 ${filepath} >> ${tmp_file}
+        else
+            echo -e ${copyright_header} >> ${tmp_file}
```

Comment:
If you change this to `>` rather than `>>` you can do away with removing the `${tmp_file}` above because then the `if` and the `else` will both create the file afresh anyway.